server/src/main/java/org/eclipse/openvsx/UpstreamRegistryService.java

```diff
@@ -252,33 +237,57 @@ public QueryResultJson query(QueryParamJson param) {
                 "targetPlatform", param.targetPlatform
         );
 
+        var queryString = queryParams.entrySet().stream()
+            .filter(entry -> !Strings.isNullOrEmpty(entry.getValue()))
+            .map(entry -> entry.getKey())
+            .map(key -> key + "={" + key + "}")
+            .collect(Collectors.joining("&"));
+        if(!queryString.isEmpty()) {
+            urlTemplate += "?" + queryString;
+        }
+
         try {
-            return restTemplate.getForObject(requestUrl, QueryResultJson.class);
+            return restTemplate.getForObject(urlTemplate, QueryResultJson.class, queryParams);
         } catch (RestClientException exc) {
-            logger.error("GET " + requestUrl, exc);
+            if(!isNotFound(exc)) {
+                var url = UriComponentsBuilder.fromUriString(urlTemplate).build(queryParams);
+                logger.error("GET " + url, exc);
+            }
             throw new NotFoundException();
         }
     }
 
     @Override
     public QueryResultJson queryV2(QueryParamJsonV2 param) {
-        try {
-            String requestUrl = createApiUrl(upstreamUrl, "api", "v2", "-", "query");
-            requestUrl = addQuery(requestUrl,
-                    "namespaceName", param.namespaceName,
-                    "extensionName", param.extensionName,
-                    "extensionVersion", param.extensionVersion,
-                    "extensionId", param.extensionId,
-                    "extensionUuid", param.extensionUuid,
-                    "namespaceUuid", param.namespaceUuid,
-                    "includeAllVersions", param.includeAllVersions,
-                    "targetPlatform", param.targetPlatform
-            );
+        var urlTemplate = urlConfigService.getUpstreamUrl() + "/api/v2/-/query";
+        var queryParams = Map.of(
+                "namespaceName", param.namespaceName,
+                "extensionName", param.extensionName,
+                "extensionVersion", param.extensionVersion,
+                "extensionId", param.extensionId,
+                "extensionUuid", param.extensionUuid,
+                "namespaceUuid", param.namespaceUuid,
+                "includeAllVersions", String.valueOf(param.includeAllVersions),
+                "targetPlatform", param.targetPlatform
+        );
 
-            return restTemplate.getForObject(requestUrl, QueryResultJson.class);
+        var queryString = queryParams.entrySet().stream()
+                .filter(entry -> !Strings.isNullOrEmpty(entry.getValue()))
+                .map(entry -> entry.getKey())
+                .map(key -> key + "={" + key + "}")
```

Comment:
Shouldn't this be `.map(key -> key + "={" + value + "}")`

Reply:
No, the key is used so that the template is uniform. This is to group all `/api/-/query` requests together. RestTemplate uses `queryParams` to fill in the template before making the request.